modules/calendar/spec/features/calendar_sharing_spec.rb

```diff
@@ -59,58 +61,109 @@
            ])
   end
 
+  shared_let(:admin) do
+    create(:admin,
+           member_in_project: project)
+  end
+
   let(:saved_query) do
     create(:query_with_view_work_packages_calendar,
            user: user_with_sharing_permission,
            project:,
            public: false)
   end
 
-  context 'with sufficient permissions', with_settings: { ical_enabled: true } do
+  # context 'without sufficient permissions and the ical_enabled setting enabled', with_settings: { ical_enabled: true } do
+  #   let(:saved_query) do
+  #     create(:query_with_view_work_packages_calendar,
+  #            user: user_without_sharing_permission,
+  #            project:,
+  #            public: false)
+  #   end
+
+  #   before do
+  #     login_as user_without_sharing_permission
+  #     calendar.visit!
+  #   end
+
+  #   context 'on persisted calendar query' do
+  #     before do
+  #       saved_query
+
+  #       visit project_calendars_path(project)
+
+  #       within '#content' do
+  #         click_link saved_query.name
+  #       end
+
+  #       loading_indicator_saveguard
+  #     end
+
+  #     it 'shows disabled sharing menu item' do
+  #       # wait for settings button to become visible
+  #       expect(page).to have_selector("#work-packages-settings-button")
+
+  #       # click on settings button
+  #       page.find_by_id('work-packages-settings-button').click
+
+  #       # expect disabled sharing menu item
+  #       within "#settingsDropdown" do
+  #         # expect(page).to have_button("Subscribe to iCalendar", disabled: true) # disabled selector not working
+  #         expect(page).to have_selector(".menu-item.inactive", text: "Subscribe to iCalendar")
+  #         page.click_button("Subscribe to iCalendar")
+
+  #         # modal should not be shown
+  #         expect(page).not_to have_selector('.spot-modal--header', text: "Subscribe to iCalendar")
+  #       end
+  #     end
+  #   end
+  # end
+
+  context 'with sufficient permissions and the ical_enabled setting enabled', with_settings: { ical_enabled: true } do
```

Comment:
Hey @mrmir, this line still has the `with_settings: { ical_enabled: true }` meta info for rspec which will lead to the setting being mocked. The scope of this context includes the specs where you have been trying to alter the setting. Removing it will lead to the specs passing.